draft-ietf-jsonpath-base.md

```diff
@@ -1036,30 +1036,33 @@ The current item is selected if and only if the result is `true`.
 
 
 ~~~~ abnf
-
-
-boolean-expr = logical-expr
-logical-expr = ([neg-op] primary-expr) / logical-or-expr
-neg-op       = "!"                                  ; not operator
-primary-expr = "(" logical-or-expr ")"
-logical-or-expr = logical-and-expr *["||" logical-and-expr]
-logical-and-expr = comp-expr *["&&" comp-expr]
-
-comp-expr    = (rel-path-val /
-                json-path) [(comp-op comparable / ; comparison
-                            regex-op regex     /  ; RegEx test
-                            in-op container )]    ; containment test
-comp-op      = "==" / "!=" /                        ; comparison ...
-               "<"  / ">"  /                        ; operators
+boolean-expr     = logical-or-expr
+logical-or-expr  = logical-and-expr *("||" logical-and-expr)
+                                                      ; disjunction
+                                                      ; binds less tightly than conjunction
+logical-and-expr = basic-expr *("&&" basic-expr)      ; conjunction
+                                                      ; binds more tightly than disjunction
+
+basic-expr   = comp-expr / existence-expr / paren-expr / (neg-op paren-expr)
+existence-expr = [neg-op] path                        ; path existence or non-existence
+paren-expr   = "(" boolean-expr ")"                   ; parenthesized expression
+neg-op       = "!"                                    ; not operator
+
+comp-expr    = path (comp-op comparable /             ; comparison
```

Comment:
@goessner I'd like to support `comparable comp-op path` as well since it seems arbitrary to require the path to be on the left hand side of the comparison operator. Something for a future PR, perhaps?

Reply:
I raised https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/112 to cover this.